fbu-app/ViewControllers/Setup/RulesViewController.m

```diff
@@ -42,39 +55,36 @@ - (void) fetchRules {
     self.rules = [self.house objectForKey:@"rules"];
 }
 
-- (void)initViewController {
-    UIViewController * timelineViewController = [[UIViewController alloc] init];
-    [self presentViewController:timelineViewController animated:YES completion:nil];
-}
-
-- (void)initTableView {
-    
-    self.tableView = [[UITableView alloc] initWithFrame:self.view.bounds];
-    
-    self.tableView.dataSource = self;
-    self.tableView.delegate = self;
-    
-    [self.tableView registerClass:[RuleCell class] forCellReuseIdentifier:@"RuleCell"];
-    
-    [self.view addSubview:self.tableView];
-}
 
 - (nonnull UITableViewCell *)tableView:(nonnull UITableView *)tableView cellForRowAtIndexPath:(nonnull NSIndexPath *)indexPath {
     
-    static NSString *cellIdentifier = @"RuleCell";
     RuleCell *cell = [tableView dequeueReusableCellWithIdentifier:@"RuleCell"];
-    
-    cell = [[RuleCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:cellIdentifier];
-    
+        
     cell.ruleLabel.text = self.rules[indexPath.row];
-    
+    cell.numberLabel.text = [NSString stringWithFormat: @"%ld", (long)(indexPath.row+1)];
     return cell;
 }
 
 - (NSInteger)tableView:(nonnull UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
     return self.rules.count;
 }
 
+- (BOOL)tableView:(UITableView *)tableView canEditRowAtIndexPath:(NSIndexPath *)indexPath {
+    return YES;
+}
 
+- (void)tableView:(UITableView *)tableView commitEditingStyle:(UITableViewCellEditingStyle)editingStyle forRowAtIndexPath:(NSIndexPath *)indexPath {
+    if (editingStyle == UITableViewCellEditingStyleDelete) {
+        [self.rules removeObjectAtIndex:indexPath.row];
+        [self reloadView];
+    }
+}
 
+- (IBAction)tapAddRule:(id)sender {
+    [self.house addUniqueObject:self.ruleField.text forKey:@"rules"];
+    [self.house save];
```

Comment:
using saveInBackground would take care of thread warnings

Reply:
got it thanks!